apollo-router/src/spec/query.rs

```diff
@@ -1056,6 +1141,42 @@ pub(crate) struct Variable {
 }
 
 impl Operation {
+    fn from_hir(operation: &hir::OperationDefinition, schema: &Schema) -> Result<Self, SpecError> {
+        let name = operation.name().map(|s| s.to_owned());
+        let kind = operation.operation_ty().into();
+        let current_field_type = match kind {
+            OperationKind::Query => FieldType::Named("Query".to_string()),
```

Comment:
now that i'm looking at it (and this appears also in the AST based version), what happens if the query type is not named `Query` here?

Reply:
Excellent question! What happens (on `dev`) is this:

```rust
thread 'spec::query::tests::test_query_not_named_query' panicked at 
'called `Result::unwrap()` on an `Err` value: InvalidField("example", "Query")', 
apollo-router/src/spec/query/tests.rs:5264:63
```

I’ll make a separate PR for this.

Reply:
right. I thought we had fixed issues with custom query objects, but apparently they were not all fixed

Reply:
https://github.com/apollographql/router/pull/2459